lib/src/api/opt/mod.rs

```diff
@@ -221,25 +236,57 @@ fn into_json(value: Value) -> serde_json::Result<JsonValue> {
-		Value::Strand(strand) => Ok(strand.0.into()),
-		Value::Duration(d) => serde_json::to_value(d),
+		Value::Strand(strand) => match simplify {
+			true => Ok(strand.0.into()),
+			false => serde_json::to_value(strand),
+		},
+		Value::Duration(d) => match simplify {
+			true => Ok(d.to_string().into()),
+			false => serde_json::to_value(d),
+		},
 		Value::Datetime(d) => serde_json::to_value(d),
 		Value::Uuid(uuid) => serde_json::to_value(uuid),
-		Value::Array(arr) => Ok(JsonValue::Array(Array::try_from(arr)?.0)),
-		Value::Object(obj) => Ok(JsonValue::Object(Object::try_from(obj)?.0)),
+		Value::Array(arr) => Ok(JsonValue::Array(Array::try_from((arr, simplify))?.0)),
+		Value::Object(obj) => Ok(JsonValue::Object(Object::try_from((obj, simplify))?.0)),
 		Value::Geometry(geometry) => serde_json::to_value(geometry),
 		Value::Bytes(bytes) => serde_json::to_value(bytes),
 		Value::Param(param) => serde_json::to_value(param),
 		Value::Idiom(idiom) => serde_json::to_value(idiom),
 		Value::Table(table) => serde_json::to_value(table),
-		Value::Thing(thing) => serde_json::to_value(thing),
+		Value::Thing(thing) => match simplify {
+			true => Ok(thing.to_string().into()),
+			false => serde_json::to_value(thing),
+		},
 		Value::Model(model) => serde_json::to_value(model),
 		Value::Regex(regex) => serde_json::to_value(regex),
 		Value::Block(block) => serde_json::to_value(block),
 		Value::Range(range) => serde_json::to_value(range),
 		Value::Edges(edges) => serde_json::to_value(edges),
 		Value::Future(future) => serde_json::to_value(future),
-		Value::Constant(constant) => serde_json::to_value(constant),
+		Value::Constant(constant) => match simplify {
+			true => match constant {
+				Constant::MathE => Ok(std::f64::consts::E.into()),
+				Constant::MathFrac1Pi => Ok(std::f64::consts::FRAC_1_PI.into()),
+				Constant::MathFrac1Sqrt2 => Ok(std::f64::consts::FRAC_1_SQRT_2.into()),
+				Constant::MathFrac2Pi => Ok(std::f64::consts::FRAC_2_PI.into()),
+				Constant::MathFrac2SqrtPi => Ok(std::f64::consts::FRAC_2_SQRT_PI.into()),
+				Constant::MathFracPi2 => Ok(std::f64::consts::FRAC_PI_2.into()),
+				Constant::MathFracPi3 => Ok(std::f64::consts::FRAC_PI_3.into()),
+				Constant::MathFracPi4 => Ok(std::f64::consts::FRAC_PI_4.into()),
+				Constant::MathFracPi6 => Ok(std::f64::consts::FRAC_PI_6.into()),
+				Constant::MathFracPi8 => Ok(std::f64::consts::FRAC_PI_8.into()),
+				Constant::MathLn10 => Ok(std::f64::consts::LN_10.into()),
+				Constant::MathLn2 => Ok(std::f64::consts::LN_2.into()),
+				Constant::MathLog102 => Ok(std::f64::consts::LOG10_2.into()),
+				Constant::MathLog10E => Ok(std::f64::consts::LOG10_E.into()),
+				Constant::MathLog210 => Ok(std::f64::consts::LOG2_10.into()),
+				Constant::MathLog2E => Ok(std::f64::consts::LOG2_E.into()),
+				Constant::MathPi => Ok(std::f64::consts::PI.into()),
+				Constant::MathSqrt2 => Ok(std::f64::consts::SQRT_2.into()),
+				Constant::MathTau => Ok(std::f64::consts::TAU.into()),
+			},
```

Comment:
Would it be possible to `impl Constant { pub(crate) fn value(&self) -> f64 { /* big match statement */ }` and then use it both here: https://github.com/surrealdb/surrealdb/blob/eeb46468aa96977edd7a3dd2e2c5b50133555a54/lib/src/sql/constant.rs#L41-L71 and in this PR? Otherwise constants are duplicated and must be maintained in two places.

Reply:
I would just say that a constant is currently just a math f64, but might not be in the future...

Reply:
There are certainly pros and cons to my proposal 👍

Note that, in the event that a small number of other constant types were added, it might be viable for `Constant::value` to return a enum of them. The match statement could also be made into a macro so that the return type of surreal vs. json value would both be supported as long as `.into()` works.

I'm happy to defer this until a future PR or abandon the idea entirely if you think the cons outweigh the pros.

Reply:
Yeah I agree with what you are saying, just something to bear in mind with the **potential** addition of non-f64 constants! I'll leave the discussion / thought process up to you and @rushmorem.

Reply:
Good catch @finnbear! I didn't realise that was duplicated there. I was jumping straight to the `Serialize` implementations. I have extracted it. I left the simple serialisation branch untouched. That will be deleted by #1881. @tobiemh Thanks for pointing it out. We can always refactor once we add the new constants. That shouldn't be a problem since `Constant::as_value()` is private.